cicerone/references/reference.py

```diff
@@ -0,0 +1,130 @@
+    description: str | None = None  # OAS 3.1+ only
+
+    @pydantic.model_validator(mode="before")
+    @classmethod
+    def handle_dollar_ref(cls, data: typing.Any) -> typing.Any:
+        """Handle $ref in input data by converting it to ref.
+
+        This allows accepting both ref= and $ref in dictionaries.
+        """
+        if isinstance(data, dict) and "$ref" in data:
+            data = data.copy()
+            data["ref"] = data.pop("$ref")
+        return data
+
+    def model_dump(self, **kwargs) -> dict[str, typing.Any]:
+        """Serialize with $ref instead of ref."""
+        result = super().model_dump(**kwargs)
+        if "ref" in result:
+            result["$ref"] = result.pop("ref")
+        return result
+
+    def __str__(self) -> str:
+        """Return a readable string representation of the reference."""
+        parts = [f"ref='{self.ref}'"]
+        if self.summary:
+            parts.append(f"summary='{self.summary[:50]}{'...' if len(self.summary) > 50 else ''}'")
+        if self.description:
+            parts.append(f"description='{self.description[:50]}{'...' if len(self.description) > 50 else ''}'")
+        return f"<Reference: {', '.join(parts)}>"
+
+    @property
+    def is_local(self) -> bool:
+        """Check if this is a local reference (starts with #)."""
+        return self.ref.startswith("#")
+
+    @property
+    def is_external(self) -> bool:
+        """Check if this is an external reference (file or URL)."""
+        return not self.is_local
+
+    @property
+    def pointer(self) -> str:
+        """Get the JSON Pointer part of the reference.
+
+        For local references like '#/components/schemas/User', returns '/components/schemas/User'.
+        For external references with fragments like 'file.yaml#/Pet', returns '/Pet'.
+        For external references without fragments, returns ''.
+        """
+        return self.ref.split("#", 1)[1] if "#" in self.ref else ""
+
+    @property
+    def document(self) -> str:
+        """Get the document part of an external reference.
+
+        For external references like './models.yaml#/Pet', returns './models.yaml'.
+        For local references, returns empty string.
+        """
+        if self.is_external:
+            return self.ref.split("#", 1)[0] if "#" in self.ref else self.ref
+        return ""
+
+    @property
+    def pointer_parts(self) -> list[str]:
+        """Get the JSON Pointer as a list of path components.
+
+        For example, '#/components/schemas/User' returns ['components', 'schemas', 'User'].
+        """
+        pointer = self.pointer
+        if not pointer or pointer == "/":
+            return []
+        return [p for p in pointer.lstrip("/").split("/") if p]
+
+    @classmethod
+    def from_dict(cls, data: dict[str, typing.Any]) -> Reference:
+        """Create a Reference from a dictionary.
+
+        Args:
+            data: Dictionary containing at least a '$ref' key
```

Comment:
Feels like if this is a condition we should raise in here is `$ref` isn't present in keys.